sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -116,6 +116,10 @@ int gnrc_lorawan_parse_dl(gnrc_lorawan_t *mac, uint8_t *buf, size_t len,
         pkt->fopts.iol_base = buf;
         pkt->fopts.iol_len = fopts_length;
         buf += fopts_length;
+        if (buf >= p_mic) {
```

Comment:
is this not already covered by the `if (buf < p_mic)`check below?

Reply:
I feel stupid as I am not seeing your argument? Could you elaborate?

Reply:
`buf >= p_mic` is the same as `buf < p_mic`, there is also a check for `fopts_length` in there (but only when port is 0?), so this code path is already (almost?) covered.

Reply:
Sorry got confused myself, I there was an `else` branch to the `if (buf < p_mic)` I guess `buf >= p_mic` is only a problem in the `fopts_length` case then? (arg, this code could need some comments)

Reply:
None at all